src/app/trips/trip-list/trip-list.component.ts

```diff
@@ -59,4 +77,64 @@ export class TripListComponent {
     this.onParamsChange();
   }
 
+  onDateSelect() {
+    this.updateDateParams();
+    this.onParamsChange();
+  }
+
+  onPreviousSelect() {
+    this.resetDateModels();
+
+    const yesterday = new Date();
+    yesterday.setDate(yesterday.getDate() - 1);
+    this.endDateToModel = {year: yesterday.getFullYear(), month: yesterday.getMonth() + 1, day: yesterday.getDate()};
+
+    this.updateDateParams();
+    this.onParamsChange();
+  }
+
+  onCurrentSelect() {
+    this.resetDateModels();
+
+    const today = new Date();
+    this.startDateToModel = {year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate()};
+    this.endDateFromModel = {year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate()};
+
+    this.updateDateParams();
+    this.onParamsChange();
+  }
+
+  onUpcomingSelect() {
+    this.resetDateModels();
+
+    const tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    this.startDateFromModel = {year: tomorrow.getFullYear(), month: tomorrow.getMonth() + 1, day: tomorrow.getDate()};
```

Comment:
Radau tokį dalyką, ant kurio daug kas pasimautų... 

Pabandžiau pasileisti ir pasižiūrėti, kaip veikia, ir pastebėjau, kad man neveikia va šitas pasirinkimas. Paspaudus nieko nevyksta. Tada pasižiūrėjau, kad čia nustatyta data kažkodėl gaunasi 2019-04-00 - neteisinga, ir krenta erroras. Žiūrėjau valandą, kas čia gali būti blogai, nu viskas atrodo gerai.
Buvo tomorrow.getDay(). O pasirodo getDay grąžina ne dienos numerį mėnesyje, o savaitės dieną (nuo 0 (sekmadienis) iki 6 (šeštadienis) https://www.w3schools.com/jsref/jsref_getday.asp
Ir toks sutapimas šią savaitę, kad pvz 5 dieną penktadienis, ir panašiai, dėl to šią savaitę getDay pilnai sutapo su getDate, kuris grąžina mėnesio dieną. O imant rytojaus dieną getDay() grąžina 0, nes rytoj sekmadienis.

Tuo tarpu, kai visose kitose kalbose būtų pavadinta getWeekNumber ar panašiai, JS sugalvojo pavadinti getDay ir  gaišti valandas developeriams ieškant klaidų 

Epic :smiley:

Iškart sutvarkiau jau pats.